### PR TITLE
refactor(theme-chalk): scss css var namespace

### PR DESCRIPTION
**Pull request by YunYouJun**
- rename `--el` css var by `getCssVar` to custom namespace
- add `cssVar` & `cssVarBlockName` to rewrite `button` `rate` css vars in styles

---

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/hooks/use-namespace/index.ts [hunk 1]
```diff
@@ -56,6 +56,29 @@ export const useNamespace = (block: string) => {
     const state = args.length >= 1 ? args[0]! : true
     return name && state ? `${statePrefix}${name}` : ''
   }
+
+  // for css var
+  // --el-xxx: value;
+  const cssVar = (object: Record<string, string>) => {
```
Comment (sxzz) [inline packages/hooks/use-namespace/index.ts:62]: Maybe we can add a overload
```ts
(str: string) => `--${namespace.value}-${str}`
```

File: packages/components/button/src/button-custom.ts [hunk 2]
```diff
@@ -56,28 +58,29 @@ export function useButtonCustomStyle(props: ButtonProps) {
...
-          '--el-button-bg-color': buttonColor,
-          '--el-button-text-color': textColor,
-          '--el-button-border-color': buttonColor,
-          '--el-button-hover-bg-color': hoverBgColor,
-          '--el-button-hover-text-color': textColor,
-          '--el-button-hover-border-color': hoverBgColor,
-          '--el-button-active-bg-color': activeBgColor,
-          '--el-button-active-border-color': activeBgColor,
-        }
+          ? `var(--${namespace}-color-white)`
```
[the remark this answers is not in the document]
Comment (YunYouJun) [reply packages/components/button/src/button-custom.ts:61]: They are equivalent, but it seems to be shorter here.

[the remark this answers is not in the document]
Comment (sxzz) [reply packages/components/button/src/button-custom.ts:61]: I mean, let's make them consistent.

[the remark this answers is not in the document]
Comment (YunYouJun) [reply packages/components/button/src/button-custom.ts:61]: Done.

File: packages/components/dialog/src/use-dialog.ts [hunk 3]
```diff
@@ -41,9 +42,12 @@ export const useDialog = (
     isNumber(props.width) ? `${props.width}px` : props.width
   )
 
+  const globalConfig = useGlobalConfig('namespace')
+  const namespace = computed(() => globalConfig.value || 'el')
+
   const style = computed<CSSProperties>(() => {
     const style: CSSProperties = {}
-    const varPrefix = `--el-dialog`
+    const varPrefix = `--${namespace.value}-dialog`
```
[the remark this answers is not in the document]
Comment (YunYouJun) [reply packages/components/dialog/src/use-dialog.ts:50]: This is a bit special, it does not have `useNamespace` in its context, so it is more concise to use globalConfig directly.

[the remark this answers is not in the document]
Comment (sxzz) [reply packages/components/dialog/src/use-dialog.ts:50]: OK.

**Review by jw-foss — APPROVED** [review]

File: packages/components/button/src/button-custom.ts [hunk 4]
```diff
@@ -9,6 +9,8 @@ export function darken(color: TinyColor, amount = 20) {
 
 export function useButtonCustomStyle(props: ButtonProps) {
   const _disabled = useDisabled()
+  const ns = useNamespace('button')
+  const namespace = ns.namespace.value
```
[the remark this answers is not in the document]
Comment (YunYouJun) [reply packages/components/button/src/button-custom.ts]: Fixed.

File: packages/components/dialog/src/use-dialog.ts [hunk 5]
```diff
@@ -41,9 +42,12 @@ export const useDialog = (
     isNumber(props.width) ? `${props.width}px` : props.width
   )
 
+  const globalConfig = useGlobalConfig('namespace')
+  const namespace = computed(() => globalConfig.value || 'el')
```
Comment (sxzz) [inline packages/components/dialog/src/use-dialog.ts:46]: Can we export `defaultNamespace` and use like this?
```ts
import { defaultNamespace } from '@element-plus/hooks'
```

```suggestion
  const globalConfig = useGlobalConfig('namespace', defaultNamespace)
```

Comment (sxzz) [reply packages/components/dialog/src/use-dialog.ts:46]: Or to reduce duplicate code, we are still better off using hooks functions.

File: packages/components/avatar/src/avatar.vue [hunk 6]
```diff
@@ -47,9 +47,9 @@ const avatarClass = computed(() => {
 const sizeStyle = computed(() => {
   const { size } = props
   return isNumber(size)
-    ? ({
-        '--el-avatar-size': addUnit(size),
-      } as CSSProperties)
+    ? (ns.cssVarBlock({
+        size: addUnit(size) || '',
+      }) as CSSProperties)
```
[the remark this answers is not in the document]
Comment (YunYouJun) [reply packages/components/avatar/src/avatar.vue:52]: CSSProperties mean some css attributes. I think `Record<string, string>` for `cssVarBlock` is more appropriate.

We can set it to `CSSProperties` only when necessary.

**Pull request merged**